### DISCUSSION TITLE
Adding custom AAs

### DISCUSSION THREAD

**Discussion by nmathai-fog** [rdkit/rdkit#7907] [Development]
Hi there, 

I have just started to play around with `Chem.MolFromSequence` and I was wondering if: 
1. at this time, is there a way to add custom (non natural) AAs? 
2. relating to the above, is there a way to submit a sequence that is delimited by a character. I.e. Ala-Ala instead of AA.. this increases the scope of custom AAs which could be added as we are not limited to single characters
3. I noticed that L and D behavior only work on lowercase - is this intensional? 

![Screenshot 2024-10-09 
![Screenshot 2024-10-09 at 8 39 44 AM](https://github.com/user-attachments/assets/8a771051-1272-4ab8-81a8-9133836a2fa6)
at 8 3
![Screenshot 2024-10-09 at 8 38 54 AM](https://github.com/user-attachments/assets/7a47da4b-20a1-4bdf-9707-4c37ba27587d)
9 44 AM](https://github.com/user-attachments/assets/c666f21f-78d1-41de-a29f-f98260704b4d)

Best wishes and TIA!

**Comment by nmathai-fog** [comment]
![Screenshot 2024-10-09 at 8 21 51 AM](https://github.com/user-attachments/assets/d4c6da52-a989-43aa-8842-626e362c8297)

**Comment by nmathai-fog** [comment]
![Screenshot 2024-10-09 at 8 38 54 AM](https://github.com/user-attachments/assets/1cde9347-987e-4312-990c-96f61460e0ff)

**Comment by greglandrum** [comment]
Hi @nmathai-fog

> I noticed that L and D behavior only work on lowercase - is this intensional?

This is the intended behavior (otherwise you could not mix D and L amino acids in one seuence), but it is not documented. We'll try to fix that in a future release.

**Comment by greglandrum** [comment]
Two answer the other two questions:

> 1. at this time, is there a way to add custom (non natural) AAs?
> 2. relating to the above, is there a way to submit a sequence that is delimited by a character. I.e. Ala-Ala instead of AA.. this increases the scope of custom AAs which could be added as we are not limited to single characters

Neither of these are currently possible

**Comment by nmathai-fog** [comment]
Thank you this makes sense! :D
